v1/plugin/plugin.go

```diff
@@ -591,9 +604,13 @@ func printPreambleAndServe(srv server, m *meta, p *pluginProxy, port string, isP
 			return "", err
 		}
 	}
+	advertisedAddr, err := getAddr(grpcListenAddr)
+	if err != nil {
+		return "", err
+	}
 	resp := preamble{
 		Meta:          *m,
-		ListenAddress: addr,
+		ListenAddress: fmt.Sprintf("%v:%v", advertisedAddr, l.Addr().(*net.TCPAddr).Port),
```

Comment:
Shouldn't the listening address be separate from advertised address? 

Here `grpcListenAddr` (`--addr`) is used both as -advertised address (reported in preamble) and -listening address `net.Listen("tcp", addr)`. 
It does matter in case a qualified domain name is added (in contrast to giving an ip). Then user doesn't have explicit control over address resolution, see also description for [net.Listen](https://golang.org/pkg/net/#Listen).